sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/DownloadCompletionReceiver.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.azure.mobile.updates;
+
+import android.app.DownloadManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+/**
+ * Process download manager callbacks.
+ */
+public class DownloadCompletionReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        /*
+         * Just resume app if clicking on pending download notification as
+         * it's awkward to click on a notification and nothing happening.
+         * Another option would be to open download list.
+         */
+        String action = intent.getAction();
+        if (DownloadManager.ACTION_NOTIFICATION_CLICKED.equals(action)) {
+            Updates.getInstance().resumeApp(context);
+        }
+
+        /*
+         * Forward the download identifier to Updates for inspection when a download completes.
+         */
+        else if (DownloadManager.ACTION_DOWNLOAD_COMPLETE.equals(action)) {
+            long downloadId = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, 0);
```

Comment:
I can use negative value in case download identifier is 0, it might be possible indeed.
However I will keep >=0 check to protect against intent injection.